vou/person.py

```diff
@@ -243,101 +246,116 @@ def will_increase_dose(
-                    
+
                     if self.rng.random() > self.downward_pressure:
-                        
+
                         will_increase_dose = True
                     else:
-                        
+
                         will_increase_dose = False
-                        increase_dose_src = last_attempt_src
+                        increase_dose_src = DoseIncreaseSource.WILL_NOT_INCREASE
                 else:
-                    
+
                     will_increase_dose = False
-                    increase_dose_src = last_attempt_src 
+                    increase_dose_src = DoseIncreaseSource.WILL_NOT_INCREASE
             else:
-                
+
                 will_increase_dose = False
-                increase_dose_src = last_attempt_src
+                increase_dose_src = DoseIncreaseSource.WILL_NOT_INCREASE
 
         else:
-            
+
             will_increase_dose = False
-            increase_dose_src = last_attempt_src
-        
+            increase_dose_src = DoseIncreaseSource.WILL_NOT_INCREASE
+
         if will_increase_dose == True:
 
             # Need to create dictionary of if will_increase_dose is true, record how to increase
 
             # if first time trying to increase dose, go to primary doctor for potential increase
-            if len(self.dose_increase_record) == 0:
+            ## Also, if the last attempt to increase from primary was successful, try again with primary
+            if (len(self.dose_increase_record) == 0) | (
+                (last_attempt_src == DoseIncreaseSource.PRIMARY_DOCTOR)
+                & (last_attempt_success == True)
+            ):
 
                 # probability of primary doctor increasing
                 if self.rng.random() <= self.source_probability_primary:
-                    will_increase_dose = True 
-                    increase_dose_src = "primary_doctor"
+                    will_increase_dose = True
+                    increase_dose_src = DoseIncreaseSource.PRIMARY_DOCTOR
 
                 else:
                     will_increase_dose = False
-                    increase_dose_src = "primary_doctor"
+                    increase_dose_src = DoseIncreaseSource.PRIMARY_DOCTOR
 
             # if the last time they tried to increase, they tried from their primary and failed, try from either dealer or secondary doctor
             ### Actually, removing primary condition -- if you tried and failed at all in the last attempt, you'll have the two below choices of
             ### dealer/secondary
             ##### If your last increase attempt was successful, but it was a secondary doctor, you'll enter this situation again
-            elif (last_attempt_success == False) | (last_attempt_src == "secondary_doctor"):
+            elif (last_attempt_success == False) | (
+                last_attempt_src == DoseIncreaseSource.SECONDARY_DOCTOR
+            ):
 
                 # _____ This may not be right
```

Comment:
Do you still have any concerns about this? The logic looks right to me, but if there is some aspect of it you're still unsure about, let's discuss before merging. If not, please remove this line.